internal/driver/basenotificationresthandler.go

```diff
@@ -0,0 +1,148 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//
+// Copyright (C) 2022 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package driver
+
+import (
+	"context"
+	"encoding/xml"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/edgexfoundry/device-sdk-go/v2/pkg/models"
+	sdkModel "github.com/edgexfoundry/device-sdk-go/v2/pkg/models"
+	sdk "github.com/edgexfoundry/device-sdk-go/v2/pkg/service"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/clients/logger"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/common"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/errors"
+
+	"github.com/IOTechSystems/onvif/event"
+	"github.com/IOTechSystems/onvif/gosoap"
+
+	"github.com/gorilla/mux"
+)
+
+const (
+	OnvifEventRestPath = "onvifevent"
+	apiResourceRoute   = common.ApiBase + "/" + OnvifEventRestPath + "/{" + common.DeviceName + "}/{" + common.ResourceName + "}"
+	handlerContextKey  = "RestNotificationHandler"
+)
+
+// RestNotificationHandler handle the notification from the camera and send to async value channel
+type RestNotificationHandler struct {
+	service     *sdk.DeviceService
+	logger      logger.LoggingClient
+	asyncValues chan<- *models.AsyncValues
+}
+
+// NewRestNotificationHandler create a new RestNotificationHandler entity
+func NewRestNotificationHandler(service *sdk.DeviceService, logger logger.LoggingClient, asyncValues chan<- *models.AsyncValues) *RestNotificationHandler {
+	handler := RestNotificationHandler{
+		service:     service,
+		logger:      logger,
+		asyncValues: asyncValues,
+	}
+
+	return &handler
+}
+
+// AddRoute adds route for receiving the notification from the camera
+func (handler RestNotificationHandler) AddRoute() errors.EdgeX {
+	if err := handler.service.AddRoute(apiResourceRoute, handler.addContext(deviceHandler), http.MethodPost); err != nil {
+		return errors.NewCommonEdgeX(errors.KindServerError, fmt.Sprintf("unable to add required route: %s: %s", apiResourceRoute, err.Error()), err)
+	}
+
+	handler.logger.Infof("Route %s added.", apiResourceRoute)
+	return nil
+}
+
+func (handler RestNotificationHandler) addContext(next func(http.ResponseWriter, *http.Request)) func(http.ResponseWriter, *http.Request) {
```

Comment:
Remove, should never be need per other comment.

Reply:
Removed.